packages/flutter/lib/src/widgets/app.dart

```diff
@@ -23,6 +23,32 @@ import 'widget_inspector.dart';
 
 export 'dart:ui' show Locale;
 
+/// The signature of [WidgetsApp.onLocaleChanged] callback.
+///
+/// A `LocaleChangedCallback` is responsible for computing the locale of the app's
+/// [Localizations] object when the app starts and when user changes the default
+/// locale for the device.
+///
+/// The `oldLocale` is null when the app starts. The `newLocale` is the device's
+/// locale when the app started, or the device locale the user selected after
+/// the app was started. The `supportedLocales` parameter is just the value of
+/// [WidgetApp.supportedLocales].
+typedef Locale LocaleChangedCallback(Locale oldLocale, Locale newLocale, Iterable<Locale> supportedLocales);
+
+/// The default value of the [WidgetsApp.onLocaleChanged] callback.
+///
+/// If `supportedLocales` contains `newLocale` then return it. Otherwise return
+/// the first supported locale whose [Locale.languageCode] matches the new locale's
+/// language code. Failing that, return `oldLocale`, or the first supported
+/// locale if the old locale is null (because this app was just started).
+Locale defaultLocaleChangedHandler(Locale oldLocale, Locale newLocale, Iterable<Locale> supportedLocales) {
```

Comment:
This should be a static method somewhere, not a global, so it doesn't come up when you look at the API docs.

Reply:
It's now just an implementation detail.